llvm/lib/Target/AMDGPU/SIRegisterInfo.cpp

```diff
@@ -2555,12 +2555,33 @@ bool SIRegisterInfo::eliminateFrameIndex(MachineBasicBlock::iterator MI,
                       TmpResultReg)
                   .addImm(ST.getWavefrontSizeLog2())
                   .addReg(FrameReg);
-              auto Add = BuildMI(*MBB, MI, DL, TII->get(AMDGPU::V_ADD_U32_e32),
-                                 TmpResultReg);
-              Add.addImm(Offset).addReg(TmpResultReg, RegState::Kill);
+
+              MachineInstrBuilder Add;
+              if ((Add = TII->getAddNoCarry(*MBB, MI, DL, TmpResultReg, *RS)) ==
+                  nullptr) {
```

Comment:
It would be easiest to read if we handled the positive cases where this returned an add before the fallback path.

Reply:
this case now will not arise, handling null case inside getAddNoCarry.

Reply:
I don't understand, getAddNoCarry has a failure path which you are checking for

Reply:
Now getAddNoCarry will return either of these three cases. V_ADD_U32_e32, V_MAD_U32_U24_e64 or V_ADD_CO_U32_e64

Reply:
there are couple of more test cases which are failing if i try to handle this, inside getAddNoCarry.
and the code above which is expecting nullptr will not be rechable.
else {
           MachineInstrBuilder MIB;
           if (!IsSALU) {
             if ((MIB = TII->getAddNoCarry(*MBB, MI, DL, ResultReg, *RS)) !=
                 nullptr) {